velox/vector/DecodedVector.cpp

```diff
@@ -487,41 +480,32 @@ DecodedVector::DictionaryWrapping DecodedVector::dictionaryWrapping(
     vector_size_t size) const {
   VELOX_CHECK(!isIdentityMapping_);
   VELOX_CHECK(!isConstantMapping_);
-
   VELOX_CHECK_LE(size, size_);
 
-  if (isOneLevelDictionary(wrapper)) {
-    // Re-use indices and nulls buffers.
-    return {wrapper.wrapInfo(), wrapper.nulls()};
-  } else {
```

Comment:
This change fixes the issue in https://github.com/facebookincubator/velox/issues/3300. Could you also add the unit test in #3300?